GitUI/CommandsDialogs/FormPull.cs

```diff
@@ -117,7 +117,7 @@ public FormPull(GitUICommands aCommands, string defaultRemoteBranch, string defa
 
             if (aCommands != null)
             {
-                _gitRemoteController = new GitRemoteController(Module);
+                _gitRemoteController = new GitRemoteController(() => Module);
```

Comment:
These are modal dialogs and it is not possible to change the GitModule during they lifespan. Perhaps there is a need for apply different names to stateless controllers and stateful ones.

Reply:
Indeed the dialog are unlikely to be affected by statefulness of IGitModule. However this way we ensure a consistency across API, which is IMO significantly more important.
I have been going through the codebase for a number of months and yet I have overlooked the statefulness of IGitModule, neither this omission was picked during reviews. If it wasn't for @gerhardol we'd probably get a few bug reports. 
So a chance that another contributor may not be able to understand the implications of statefulness is probably quite high.

Reply:
The problem I see is when it comes to async calls. GitModule being a function can be easily overlooked to be passed as a captured state to an async call (especially for async await constructs).